cpp/src/arrow/type.h

```diff
@@ -498,25 +498,34 @@ class ARROW_EXPORT StructType : public NestedType {
   std::vector<BufferDescr> GetBufferLayout() const override;
 };
 
-class ARROW_EXPORT Decimal128Type : public FixedSizeBinaryType {
+class ARROW_EXPORT _DecimalBaseType : public FixedSizeBinaryType {
```

Comment:
I don't think underscores are allowed by the C++ standard. How about `DecimalBaseType`?

Reply:
Sure, this is only temporary until I can remove usages of `DecimalType` from parquet-cpp